src/client/sandbox/storages/index.ts

```diff
@@ -29,22 +29,157 @@ const API_KEY_PREFIX = 'hammerhead|api-key-prefix|';
 const STORAGE_PROPS  = nativeMethods.arrayConcat.call(nativeMethods.objectKeys(Storage.prototype),
     StorageWrapper.INTERNAL_METHODS);
 
-export default class StorageSandbox extends SandboxBase {
-    private localStorageProxy: StorageProxy | null = null;
-    private sessionStorageProxy: StorageProxy | null = null;
+interface StorageSandboxStrategy {
+    localStorageProxy: StorageProxy | null;
+    sessionStorageProxy: StorageProxy | null;
+
+    attach (window: Window & typeof globalThis): void;
+
+    lock (): void;
+
+    dispose (): void;
+
+    clear (): void;
+
+    restore (storagesBackup: StoragesBackup): void;
+
+    backup (): StoragesBackup;
+}
+
+class StorageSandboxStrategyBase {
+    localStorageProxy: StorageProxy | null;
+    sessionStorageProxy: StorageProxy | null;
+}
+
+class StorageSandboxProxylessStrategy extends StorageSandboxStrategyBase implements StorageSandboxStrategy {
+    backup (): StoragesBackup {
+        return {
+            localStorage:   JSON.stringify([Object.keys(localStorage), Object.values(localStorage)]),
```

Comment:
Use the `Object.keys` method from the `nativeMethods` sandbox.

`JSON.stringify` -> `stringifyJSON`